tmt/steps/report/reportportal.py

```diff
@@ -586,53 +612,56 @@ def upload_result_logs(
         Upload all result log files into the ReportPortal instance
         """
 
-        for log_path in result.log:
-            if log_path.name == tmt.steps.execute.TEST_FAILURES_FILENAME:
-                continue
+        def upload_log(log_path: Path, is_traceback: bool = False) -> None:
             try:
-                log = self.step.plan.execute.read(log_path)
-            except tmt.utils.FileError:
-                continue
+                if log_path.suffix in ('.yaml', '.yml'):
+                    logs = tmt.utils.yaml_to_list(self.step.plan.execute.read(log_path))
+                else:
+                    logs = [self.step.plan.execute.read(log_path)]
+            except (tmt.utils.FileError, tmt.utils.GeneralError) as error:
+                tmt.utils.show_exception_as_warning(
+                    exception=error,
+                    message=f"Failed to read log file '{log_path}' for ReportPortal upload.",
+                    logger=self._logger,
+                )
+                return
 
-            message = _filter_log(log, settings=LogFilterSettings(size=self.data.log_size_limit))
-
-            # Upload log
-            self.rp_api_post(
-                session=session,
-                path="log/entry",
-                json={
-                    "message": message,
-                    "itemUuid": item_uuid,
-                    "launchUuid": launch_uuid,
-                    "level": "INFO",
-                    "time": timestamp,
-                },
+            filter_settings = LogFilterSettings(
+                size=self.data.traceback_size_limit if is_traceback else self.data.log_size_limit,
+                is_traceback=is_traceback,
             )
 
-        # Optionally write out failures
+            for log in logs:
+                # Upload log
+                self.rp_api_post(
+                    session=session,
+                    path="log/entry",
+                    json={
+                        "message": _filter_log(log, filter_settings),
+                        "itemUuid": item_uuid,
+                        "launchUuid": launch_uuid,
+                        "level": "ERROR" if is_traceback else "INFO",
+                        "time": timestamp,
+                    },
+                )
+
+        # Upload result logs
+        for log_path in result.log:
+            if any(re.search(name, log_path.name) for name in self.data.log):
```

Comment:
`if any(pattern.search(log_path.name)) for pattern in self.data.log`, because we would have patterns already compiled.